src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -41,4 +44,31 @@ public void arcadeDrive(double move, double rot) {
   public void tankDrive(double left, double right) {
     m_drive.tankDrive(left, right);
   }
+
+  public double getDistance() {
+    // Calculates the average wheel sensor position(for example, if the right wheels did a one full rotation, and the left wheels did a 50% rotation, averageWheels will be 0.75 rotation)
+    double averageWheels = (m_leftFront.getSelectedSensorPosition() + m_leftRear.getSelectedSensorPosition() + m_rightFront.getSelectedSensorPosition() + m_rightRear.getSelectedSensorPosition()) / 4 / Constants.DriveTrainConstants.kMotorToWheelRatio / Constants.DriveTrainConstants.kEncoderResolution; 
+    // the average wheels position * Wheel circumference = overall passed distance
+    return averageWheels * Constants.DriveTrainConstants.kWheelCircumference;
+  }
+
+  public void resetEncoders() {
+    m_leftFront.setSelectedSensorPosition(0);
+    m_leftRear.setSelectedSensorPosition(0);
+    m_rightFront.setSelectedSensorPosition(0);
+    m_rightRear.setSelectedSensorPosition(0);
+  }
+
+  public double getLeftWheelsRPM() {
+    return ((m_leftFront.getSelectedSensorVelocity() * Constants.DriveTrainConstants.k100msTo60sRatio) / 2048 + (m_leftRear.getSelectedSensorVelocity() * Constants.DriveTrainConstants.k100msTo60sRatio / 2048)) / 2 / 4;
+  }
+
+  public double getRightWheelsRPM() {
+    return ((m_rightFront.getSelectedSensorVelocity() * Constants.DriveTrainConstants.k100msTo60sRatio) / 2048 + (m_rightRear.getSelectedSensorVelocity() * Constants.DriveTrainConstants.k100msTo60sRatio / 2048)) / 2 / 4;
+  }
+
+  public double getAverageWheelsRPM() {
+    return (getLeftWheelsRPM() + getRightWheelsRPM()) / 4;
```

Comment:
Why divide by **4**? It is true you have four encoders but you calculate average of only two values, one is the average RPM of the left wheels and the other is the average RPM of the right wheels.

Reply:
Oops, at the time I probably thought of the motor to wheel ratio or idk, will fix